.github/workflows/ci-libshortfin.yml

```diff
@@ -36,17 +36,48 @@ env:
 
 jobs:
   build-and-test:
-    name: Build and test
-    runs-on: ubuntu-24.04
+    name: "Unit tests :: ${{ matrix.name }} :: ${{ matrix.python-version }}"
+    runs-on: ${{ matrix.runs-on }}
+    defaults:
+      run:
+        shell: bash
     strategy:
+      fail-fast: false
       matrix:
```

Comment:
If we want to make a combination of these a "required check", we could use one of these strategies from IREE:
* summary step: https://github.com/iree-org/iree/blob/3129fa924cfd5ebc581247df95789eb75a190bf0/.github/workflows/ci.yml#L291-L297
* `continue-on-error` with a "experimental" setting for jobs that are allowed to fail in the matrix: https://github.com/iree-org/iree/blob/3129fa924cfd5ebc581247df95789eb75a190bf0/.github/workflows/build_package.yml#L115

We could also just list the individual checks as required, but that makes refactoring workflows harder (since if a workflow changes on `main` then PRs synced before then will be stuck waiting for those checks).
![image](https://github.com/user-attachments/assets/2767a34e-bf36-4040-a47d-984aba1ffbec)